changelog.d/19595_lock_data_dir.fix.md

```diff
@@ -0,0 +1 @@
+Acquire and hold an exclusive lock to the global data directory until the config is dropped.
```

Comment:
```suggestion
Vector now acquires an exclusive lock to the data directory. This avoids two Vector processes accidentally using the same data directory.
```

Users aren't going to know what "until the config is dropped" means. For these changelog fragments, it's good to keep in mind that the audience is Vector users, rather than Vector developers.